### DISCUSSION TITLE
WASM build/runs for the Rust implementation

### DISCUSSION THREAD

**Discussion by tracker1** [PlummersSoftwareLLC/Primes#223] [Ideas]
Per some discussion on #29 it would be very cool to see a WASM build of the Rust implementation with tests in a couple of different runtimes, Wasmer, Node and Deno likely being the most common.

**Comment by marghidanu** [comment]
There's a merge request for an implementation is AssemblyScript. But The Rust WASM could work as well :)

**Comment by mike-barber** [comment]
That's a really cool idea; it'll be fun to see how it compares with the AssemblyScript too. 

Big question I have is where to put it since it's kind of Rust and kind of WASM. 
- Would it go in the Rust folder? 
- Would it make sense to have a WASM folder? 

We could re-use some of my Rust implementation in `solution_1` for this, but I think it'd make sense to have a more focused project that's easier to grok. And I'm far from the right person to setup all the JS/WASM side of things :P

**Comment by marghidanu** [comment]
Hmm, don't know what to say about having a WASM folder. I would suggest keeping it under Rust.

**Comment by tracker1** [comment]
Might be best to have a wasm_runtimes directory, with appropriate config, where you can have the runtime configs for wasm run from the appropriate source/build path(s) ... though not sure in either case.
